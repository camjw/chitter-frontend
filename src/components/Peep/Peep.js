import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import LikeButton from '../LikeButton'

const Peep = ({ peep }) => (
  <div className="container">
    <div className="row">
      <h4>{peep.body}</h4>
    </div>
    <div className="row">
      <div className="col-sm-4">
          Peeped by
        {' '}
        {peep.user.handle}
        {' '}
        {moment(peep.created_at).fromNow()}
  .
      </div>
      <div className="col-sm-4">
        {peep.likes.length}
        {' '}
        {peep.likes.length === 1 ? 'like' : 'likes'}
      </div>
      <div className="col-sm-4">
        <LikeButton />
      </div>
    </div>
    <hr />
    <br />
  </div>
);

Peep.propTypes = {
  peep: PropTypes.shape().isRequired,
};

export default Peep;
