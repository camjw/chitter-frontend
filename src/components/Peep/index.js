import Peep from './Peep'

export default Peep
