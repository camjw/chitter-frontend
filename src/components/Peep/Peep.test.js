import React from 'react';
import { shallow } from 'enzyme'
import Peep from './Peep'

const peepWrapper = shallow(<Peep peep={ {body: 'Hello Jest'} } />)

describe('Peep', () => {
  it('renders some HTML', () => {
    expect(peepWrapper.html()).toEqual('Hello')
  })
})
