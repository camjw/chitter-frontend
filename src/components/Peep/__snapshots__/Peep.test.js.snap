// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peep #render renders likes if there are any 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><h4>Hello Jest</h4></div><div class=\\"row\\"><div class=\\"col-sm-4\\">Peeped by Test user a few seconds ago.</div><div class=\\"col-sm-4\\">1 like</div><div class=\\"col-sm-4\\"><button>Like</button></div></div><hr/><br/></div>"`;

exports[`Peep #render renders some HTML 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><h4>Hello Jest</h4></div><div class=\\"row\\"><div class=\\"col-sm-4\\">Peeped by Test user a few seconds ago.</div><div class=\\"col-sm-4\\">0 likes</div><div class=\\"col-sm-4\\"><button>Like</button></div></div><hr/><br/></div>"`;
